### PR TITLE
Add endpointAttachment in Apigee

### PR DESCRIPTION
**Pull request by xuchenma**
This PR adds a new resource EndpointAttachment in Apigee (https://github.com/hashicorp/terraform-provider-google/issues/11118)

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_apigee_endpoint_attachment`
```

### CODE REVIEW

**Comment by xuchenma** [comment]
I ran the Terraform test case locally, and it was able to CREATE, GET, DELETE this resource. However the test returns this error:
```
=== RUN   TestAccApigeeEndpointAttachment_apigeeEndpointAttachmentBasicTestExample
=== PAUSE TestAccApigeeEndpointAttachment_apigeeEndpointAttachmentBasicTestExample
=== CONT  TestAccApigeeEndpointAttachment_apigeeEndpointAttachmentBasicTestExample
    provider_test.go:278: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
        
        (map[string]string) (len=4) {
         (string) (len=4) "host": (string) "",
         (string) (len=8) "location": (string) "",
         (string) (len=4) "name": (string) "",
         (string) (len=18) "service_attachment": (string) ""
        }
        
        
        (map[string]string) (len=4) {
         (string) (len=4) "host": (string) (len=7) "7.0.0.2",
         (string) (len=8) "location": (string) (len=8) "us-west2",
         (string) (len=4) "name": (string) (len=56) "organizations/apigee-tf-test-2/endpointAttachments/test1",
         (string) (len=18) "service_attachment": (string) (len=72) "projects/apigee-tf-test-2/regions/us-west2/serviceAttachments/my-psc-ilb"
        }
--- FAIL: TestAccApigeeEndpointAttachment_apigeeEndpointAttachmentBasicTestExample (645.56s)

```
I'm not sure what might go wrong here.

File: mmv1/products/apigee/terraform.yaml [hunk 1]
```diff
@@ -218,6 +218,31 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
+    examples:
+    - !ruby/object:Provider::Terraform::Examples
+      name: "apigee_endpoint_attachment_basic"
+      skip_test: true
+    - !ruby/object:Provider::Terraform::Examples
+      # This is a more verbose version of the above that creates all
+      # the resources needed for the acceptance test.
+      name: "apigee_endpoint_attachment_basic_test"
+      primary_resource_id: "apigee_endpoint_attachment"
+      test_env_vars:
```
Comment (c2thorn) [inline mmv1/products/apigee/terraform.yaml:235]: will also need a `vars` block that defines variable names for each resource name. See https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/products/pubsublite/terraform.yaml#L51

File: mmv1/templates/terraform/examples/apigee_endpoint_attachment_basic.tf.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+data "google_client_config" "current" {}
+
+resource "google_compute_network" "apigee_network" {
+  name = "apigee-network"
```
Comment (c2thorn) [inline mmv1/templates/terraform/examples/apigee_endpoint_attachment_basic.tf.erb:4]: Each resource name should be wrapped in a ruby tag, see https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/templates/terraform/examples/pubsub_lite_subscription_basic.tf.erb

**Review by c2thorn — CHANGES_REQUESTED** [review]
I am still working on reproducing your issue, but running into difficulties with rate limiting in our testing environment.
While that is pending, I can immediately identify an issue: the resource names in your example file will need to be wrapped in ruby tags and referenced in the `terraform.yaml`.
This is for multiple reasons, but the primary one being so that we can append a random suffix to resource names to avoid collisions while testing our entire suite.

**Comment by xuchenma** [comment]
That might not be needed for Apigee resources, because for Apigee, each test creates its own GCP project, run the test, and delete the project. So collisions won't happen. I see tests for other Apigee resources are all written this way (no ruby tags).

**Comment by xuchenma** [comment]
I updated `mmv1/templates/terraform/custom_import/apigee_endpoint_attachment.go.erb` and it fixed the `ImportStateVerify` error; the tests are passing now.

**Comment by c2thorn** [comment]
VCR test passed from what is available, but apparently we need to skip VCR for the main test. Running the primary example test directly in our testing environment

**Review by c2thorn — APPROVED** [review]
Implementation LGTM, noting we still need to have a passing test, which is running now.

**Comment by c2thorn** [comment]
manual test succeeded!

**Pull request merged**